Compile error in `lib/vector-core/buffers/examples/soak.rs`

I'm getting a compile error when checking the above source with `cargo check --all-targets --no-default-features --workspace`:

```
error[E0599]: no variant named `Disk` found for enum `Variant`
   --> lib/vector-core/buffers/examples/soak.rs:102:28
    |
102 |     let variant = Variant::Disk {
    |                            ^^^^ variant not found in `Variant`

```

If we add a `#if` guard to that variant, we'll have to eliminate the whole example, but I'm not sure if this is the right approach, or if something else could still be used there for an example.

--- a/lib/vector-core/buffers/examples/soak.rs
+++ b/lib/vector-core/buffers/examples/soak.rs
@@ -1,4 +1,6 @@
-use std::{env, fmt, path::PathBuf, process, time::Duration};
+#[cfg(feature = "disk-buffer")]
+use std::path::PathBuf;
+use std::{env, fmt, process, time::Duration};
 
 use buffers::{
     bytes::{DecodeBytes, EncodeBytes},
@@ -91,18 +93,26 @@ async fn main() {
         .expect("exporter install should not fail");
 
     let _ = cli_args.remove(0);
-    let data_dir: PathBuf = cli_args
-        .remove(0)
-        .parse()
-        .expect("database path must be a valid path");
-    let db_size: usize = cli_args
-        .remove(0)
-        .parse()
-        .expect("database size must be a non-negative amount");
-    let variant = Variant::Disk {
-        id: "debug".to_owned(),
-        data_dir,
-        max_size: db_size,
+    #[cfg(feature = "disk-buffer")]
+    let variant = {
+        let data_dir: PathBuf = cli_args
+            .remove(0)
+            .parse()
+            .expect("database path must be a valid path");
+        let db_size: usize = cli_args
+            .remove(0)
+            .parse()
+            .expect("database size must be a non-negative amount");
+        Variant::Disk {
+            id: "debug".to_owned(),
+            data_dir,
+            max_size: db_size,
+            when_full: WhenFull::DropNewest,
+        }
+    };
+    #[cfg(not(feature = "disk-buffer"))]
+    let variant = Variant::Memory {
+        max_events: 99999,
         when_full: WhenFull::DropNewest,
     };
 

--- a/lib/vector-core/buffers/src/lib.rs
+++ b/lib/vector-core/buffers/src/lib.rs
@@ -37,6 +37,7 @@ pub use variant::*;
 ///
 /// This function will fail only when creating a new disk buffer. Because of
 /// legacy reasons the error is not a type but a `String`.
+#[allow(clippy::needless_pass_by_value)]
 pub fn build<'a, T>(
     variant: Variant,
 ) -> Result<

--- a/lib/vector-core/buffers/src/variant/memory_only.rs
+++ b/lib/vector-core/buffers/src/variant/memory_only.rs
@@ -2,7 +2,7 @@ use crate::WhenFull;
 #[cfg(test)]
 use quickcheck::{Arbitrary, Gen};
 
-#[derive(Debug, Clone, Copy)]
+#[derive(Debug, Clone)]
 pub enum Variant {
     Memory {
         max_events: usize,

